### PR TITLE
Add comprehensive test coverage for EXEC prepare method

### PR DESCRIPTION
**Pull request by Ananya2**
## Overview

The `PrepareMethodExecTest.java` test suite provides comprehensive validation for the EXEC prepare method implementation in the Microsoft SQL Server JDBC Driver. This implementation specifically addresses **Sybase ASE compatibility** requirements, enabling seamless migration from Sybase environments to SQL Server by maintaining session-scoped temporary table persistence across PreparedStatement boundaries.

## Test Strategy

The testing strategy follows a multi-layered approach covering:

1. **Core Functionality Validation** - Primary Sybase ASE compatibility requirements
2. **Comprehensive Data Type Testing** - All supported SQL data types with edge cases
3. **Security Validation** - SQL injection prevention across attack vectors
4. **Performance Assessment** - Comparative analysis with existing prepare methods
5. **Edge Case Handling** - Boundary conditions and error scenarios
6. **Real-world Migration Scenarios** - Complex Sybase migration patterns

---

#### **Scenario 1.1: Temporary Table Persistence**
- **Test Method**: `testTempTablePersistenceWithExecMethod()`
- **Business Logic**: Validates that temporary tables created in one PreparedStatement remain accessible across different PreparedStatements within the same session
- **Impact**: **Critical** - This is the primary differentiator for Sybase migration. Traditional `prepare` and `prepexec` methods would fail this scenario
- **Test Flow**:
  1. Create temporary table using EXEC method
  2. Insert data using different PreparedStatement
  3. Query data using third PreparedStatement  
  4. Update data using fourth PreparedStatement
  5. Verify persistence across all operations
- **Validation**: Ensures temp tables persist in session scope, enabling complex Sybase migration scenarios

#### **Scenario 1.2: Complex Multi-Table Operations**
- **Test Method**: `testMultipleTempTablesScenario()`
- **Business Logic**: Simulates real-world Sybase migration scenarios involving multiple temporary tables with JOIN operations
- **Impact**: **High** - Validates complex migration patterns common in Sybase applications
- **Test Flow**:
  1. Create multiple temporary tables (#orders, #customers)
  2. Insert data into both tables using separate PreparedStatements
  3. Execute complex JOIN queries across temporary tables
  4. Verify data integrity and accessibility
- **Validation**: Confirms that EXEC method supports complex temporary table workflows essential for Sybase migration

---

#### **Scenario 2.1: Basic SQL Injection Prevention**
- **Test Method**: `testSQLInjectionPrevention()`
- **Business Logic**: Validates that EXEC method properly escapes parameters to prevent SQL injection attacks
- **Impact**: **Critical** - Ensures security parity with existing prepare methods
- **Attack Vectors Tested**:
  - Classic injection: `'; DROP TABLE users; --`
  - Boolean-based: `' OR '1'='1`
  - Comment-based: `admin'/*`
  - Command execution: `1'; EXEC xp_cmdshell('dir'); --`
- **Validation**: All injection attempts return as literal strings, not executed code

#### **Scenario 2.2: Advanced Attack Pattern Prevention**
- **Test Method**: `testAdvancedSQLInjectionPrevention()`
- **Business Logic**: Tests sophisticated attack patterns that bypass basic filters
- **Impact**: **High** - Protects against advanced persistent threats
- **Advanced Patterns Tested**:
  - Time-based attacks: `'; WAITFOR DELAY '00:00:05'; --`
  - UNION-based attacks: `' UNION SELECT TOP 1 name FROM sys.tables --`
  - System procedure attacks: `'; EXEC sp_configure 'show advanced options', 1; --`
  - Information extraction: `' AND (SELECT COUNT(*) FROM sys.tables) > 0 --`
  - Variable declaration attacks: SQL variable manipulation attempts
- **Validation**: Ensures robust protection against enterprise-level security threats

---

#### **Scenario 3.1: Multi-Data Type Parameter Substitution**
- **Test Method**: `testAllDataTypesParameterSubstitution()`
- **Business Logic**: Validates parameter substitution for all supported SQL data types in a single comprehensive test
- **Impact**: **High** - Ensures complete data type compatibility for migration scenarios
- **Data Types Covered**: VARCHAR, NVARCHAR, INTEGER, DECIMAL, DATETIME, BINARY, BIT
- **Test Sets**:
  1. **Normal Values**: Standard data type values
  2. **Edge Values**: Boundary conditions (MAX_VALUE, special characters)
  3. **NULL Values**: NULL handling across all data types
  4. **Extreme Values**: Empty strings, mathematical scripts, zero values
  5. **Injection Attempts**: Security validation within data type context
- **Validation**: Confirms accurate parameter substitution and data integrity across all supported types

#### **Scenario 3.2: Numeric Precision and Scale Handling**
- **Test Method**: `testNumericPrecisionScale()`
- **Business Logic**: Validates that numeric precision and scale are preserved during parameter substitution
- **Impact**: **Medium** - Ensures financial and scientific data accuracy
- **Precision Tests**:
  - `123.456` (precision: 6, scale: 3)
  - `999999.999999` (precision: 12, scale: 6)
  - `0.000001` (precision: 7, scale: 6)
  - `123456789012345.123456789` (precision: 24, scale: 9)
- **Validation**: Confirms that decimal precision is maintained without data loss

#### **Scenario 3.3: Unicode and International Character Support**
- **Test Method**: `testUnicodeEdgeCases()`
- **Business Logic**: Validates proper handling of international characters, emojis, and special Unicode scenarios
- **Impact**: **Medium** - Essential for global applications and migration scenarios
- **Unicode Categories Tested**:
  - Emojis: 🌟🎯🔥
  - Latin with accents: Ñoño México
  - Cyrillic: Здравствуйте
  - Japanese: こんにちは
  - Mathematical scripts: 𝒪𝓃𝑒 𝒯𝓌𝑜
  - UTF-16 surrogates: Complex Unicode encoding
- **Validation**: Ensures proper character encoding and preservation across all Unicode categories

#### **Scenario 3.4: Binary Data Edge Cases**
- **Test Method**: `testBinaryEdgeCases()`
- **Business Logic**: Validates handling of binary data including full byte range and large binary objects
- **Impact**: **Medium** - Important for applications handling binary content (files, images, etc.)
- **Binary Tests**:
  - All byte values (0-255)
  - Large binary data (8KB)
  - Single byte values
  - Empty binary arrays
- **Validation**: Confirms accurate binary data preservation and retrieval

---

#### **Scenario 4.1: DateTime Precision Validation**
- **Test Method**: `testDateTimePrecision()`
- **Business Logic**: Validates microsecond precision and edge date handling
- **Impact**: **Medium** - Critical for time-sensitive applications and historical data
- **DateTime Tests**:
  - Microsecond precision: `2023-12-25 15:30:45.123456`
  - Minimum SQL Server date: `1753-01-01 00:00:00.000`
  - Maximum SQL Server date: `9999-12-31 23:59:59.997`
- **Validation**: Ensures datetime precision is preserved within SQL Server limits

#### **Scenario 4.2: Large Number Handling**
- **Test Method**: `testLargeNumbers()`
- **Business Logic**: Validates handling of very large decimal numbers and scientific notation
- **Impact**: **Low-Medium** - Important for financial and scientific applications
- **Large Number Tests**:
  - Very large decimals with high precision
  - Very small decimals (micro-values)
  - Scientific notation: `1.23E+15`
- **Validation**: Confirms numeric accuracy within SQL Server precision limits

#### **Scenario 4.3: Long String Parameter Handling**
- **Test Method**: `testLongStringParameters()`
- **Business Logic**: Validates handling of long string parameters across various lengths
- **Impact**: **Medium** - Important for applications with large text fields
- **String Lengths Tested**: 100, 1000, 4000, 8000 characters
- **Validation**: Ensures string truncation does not occur and full content is preserved

---

#### **Scenario 5.1: Comparative Performance Analysis**
- **Test Method**: `testBasicPerformanceComparison()`
- **Business Logic**: Compares performance of EXEC method against PREPEXEC and PREPARE methods
- **Impact**: **High** - Validates that new method doesn't introduce significant performance degradation
- **Performance Metrics**:
  - 50 iterations of PreparedStatement creation and execution
  - Timing comparison across all three methods
  - Performance ratio validation (EXEC should be within 5x of other methods)
- **Validation**: Ensures EXEC method is performant enough for production use

#### **Scenario 5.2: Repeated Execution Performance**
- **Test Method**: `testRepeatedExecutionPerformance()`
- **Business Logic**: Tests performance when reusing the same PreparedStatement multiple times
- **Impact**: **Medium** - Validates efficiency in high-throughput scenarios
- **Test Pattern**:
  - Single PreparedStatement with 50 executions
  - Parameter reuse across executions
  - Memory and timing efficiency validation
- **Validation**: Confirms that EXEC method efficiently handles repeated executions

---

#### **Scenario 6.1: Batch Execution Support**
- **Test Method**: `testBatchExecution()`
- **Business Logic**: Validates that EXEC method properly supports batch operations with temporary tables
- **Impact**: **Medium** - Important for bulk operations in migrated Sybase applications
- **Batch Operations**:
  - Create temporary table
  - Add multiple batch statements
  - Execute batch with `executeBatch()`
  - Verify all batch operations succeeded
- **Validation**: Ensures batch operations work correctly with temporary table persistence

#### **Scenario 6.2: NULL Value Handling**
- **Test Method**: `testNullHandlingAllTypes()`
- **Business Logic**: Validates proper NULL handling across all supported data types
- **Impact**: **Medium** - Ensures data integrity when dealing with optional/missing values
- **NULL Tests**:
  - NULL values for all major data types
  - `wasNull()` validation after retrieval
  - Mixed NULL and non-NULL scenarios
- **Validation**: Confirms proper NULL semantics are maintained

### CODE REVIEW

File: src/test/java/com/microsoft/sqlserver/jdbc/callablestatement/CallableStatementTest.java [hunk 1]
```diff
@@ -568,90 +591,89 @@ public void testCallableStatementSetByAnnotatedArgs() throws SQLException {
...
-    //             TestUtils.dropTableIfExists(table, stmt);
-    //         }
-    //     }
-    // }
+    @Test
+    @Tag(Constants.reqExternalSetup)
+    @Tag(Constants.xAzureSQLDB)
+    @Tag(Constants.xAzureSQLDW)
+    @Tag(Constants.xAzureSQLMI)
+    public void testFourPartSyntaxCallEscapeSyntax() throws SQLException {
```
Comment (muskan124947) [inline src/test/java/com/microsoft/sqlserver/jdbc/callablestatement/CallableStatementTest.java]: This test needs to be commented due to failure with sql server 2025.
